File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/enums/TradeItOrderPriceType.java
package it.trade.android.sdk.enums;


public enum TradeItOrderPriceType {
    MARKET("market"),
    LIMIT("limit"),
    STOP_MARKET("stopLimit"),
    STOP_LIMIT("stopMarket");

    private String priceTypeValue;

    TradeItOrderPriceType(String priceTypeValue) {
        this.priceTypeValue = priceTypeValue;
    }

    public String getPriceTypeValue() {
        return priceTypeValue;
    }
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/enums/TradeItOrderPriceType.kt
package it.trade.android.sdk.enums


enum class TradeItOrderPriceType private constructor(val priceTypeValue: String) {
    MARKET("market"),
    LIMIT("limit"),
    STOP_MARKET("stopMarket"),
    STOP_LIMIT("stopLimit");


    companion object {

        fun getPriceTypeForValue(actionValue: String): TradeItOrderPriceType? {
            for (priceType in TradeItOrderPriceType.values()) {
                if (priceType.priceTypeValue == actionValue) {
                    return priceType
                }
            }
            return null
        }
    }
}
